### PR TITLE
feat: make authn-app compliant with OEP-55

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### Description

Add catalog-info.yaml file. Added file according to this Doc: https://docs.openedx.org/en/latest/developers/how-tos/add-a-catalog-file.html

#### https://2u-internal.atlassian.net/browse/VAN-1204

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This file records information about this repo. Its use is described in OEP-55:
+# https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0055-proc-project-maintainers.html
+
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: 'frontend-app-authn'
```
Comment (zainab-amir) [inline catalog-info.yaml:7]:
```suggestion
  name: 'Authentication Frontend'
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  description: "Micro-frontend application responsible for the login, registration and password reset functionality."
```
Comment (zainab-amir) [inline catalog-info.yaml:8]:
```suggestion
  description: "Micro-frontend for authentication service. It contains views for login, registration and password reset functionality."
```

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  links:
+    - url: "https://github.com/openedx/frontend-app-authn"
+        title: "Source Code"
+        icon: "Code"
```
Comment (zainab-amir) [inline catalog-info.yaml:12]: Remove this

File: catalog-info.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    - url: 'https://github.com/openedx/frontend-app-authn/blob/master/README.rst'
+      title: 'Documentation'
+      icon: 'Article'
+    - url: "https://authn.edx.org/register"
+        title: "Production Site"
+        icon: "Web"
```
Comment (zainab-amir) [inline catalog-info.yaml:18]: Remove this

File: catalog-info.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  annotations:
+    openedx.org/arch-interest-groups: ""
+spec:
+  owner: group:vanguards
+  type: 'library'
+  lifecycle: 'production'
```
Comment (zainab-amir) [inline catalog-info.yaml:24]:
```suggestion
  lifecycle: 'production'
dependsOn:
- 'lms'
```

File: catalog-info.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      title: 'Documentation'
+      icon: 'Article'
+    - url: "https://authn.edx.org/register"
+        title: "Production Site"
+        icon: "Web"
+  annotations:
+    openedx.org/arch-interest-groups: ""
+spec:
+  owner: group:vanguards
+  type: 'library'
```
Comment (zainab-amir) [inline catalog-info.yaml:23]:
```suggestion
  type: 'service'
```

**Comment by attiyaIshaque** [comment]
@zainab-amir I have addressed your comments.

**Pull request merged**